source/common/network/io_socket_handle_impl.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include "envoy/network/io_handle.h"
+
+namespace Envoy {
+namespace Network {
+
+/**
+ * IoHandle derivative for sockets
+ */
+class IoSocketHandle : public IoHandle {
+public:
+  IoSocketHandle(int fd = -1) : fd_(fd) {}
+
+  // TODO(sbelair2) Call close() in destructor
+  ~IoSocketHandle() {}
+
+  // TODO(sbelair2)  To be removed when the fd is fully abstracted from clients.
+  int fd() const override { return fd_; }
+
+  void close() override { fd_ = -1; }
+
+  // TODO(sbelair2)  To be removed when the fd is fully abstracted from clients.
+  void operator=(int fd) { fd_ = fd; }
```

Comment:
Is this still needed? Please remove?

Reply:
Note that it is removed from the base class. but I still needed it for the OriginalDstTestFilter tests, but I may be able to get around it still, not sure.